pkg/operator/controller/ingress-routes/controller.go

```diff
@@ -0,0 +1,250 @@
+	log = logf.Logger.WithName(controllerName)
+)
+
+// New creates the ingress routes controller from configuration. This is the controller
+// that handles all the logic for generating roles and rolebindings for operators that
+// include routes with configurable hostnames and service certificate.
+func New(mgr manager.Manager, config Config) (controller.Controller, error) {
+	reconciler := &reconciler{
+		config: config,
+		client: mgr.GetClient(),
+	}
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: reconciler})
+	if err != nil {
+		return nil, err
+	}
+	if err := c.Watch(&source.Kind{Type: &configv1.Ingress{}}, &handler.EnqueueRequestForObject{}); err != nil {
+		return nil, err
+	}
+	// TODO: The operator only watches for events in the openshift-ingress-operator namespace,
+	// changes to generated role and rolebindings in the openshift-config namespace are ignored.
+	// Find a way to trigger events outside of the openshift-ingress-operator namespace.
+	if err := c.Watch(&source.Kind{Type: &rbacv1.Role{}}, &handler.EnqueueRequestForOwner{OwnerType: &configv1.Ingress{}}); err != nil {
+		return nil, err
+	}
+	if err := c.Watch(&source.Kind{Type: &rbacv1.RoleBinding{}}, &handler.EnqueueRequestForOwner{OwnerType: &configv1.Ingress{}}); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+// Config holds all the things necessary for the controller to run.
+type Config struct {
+	SecretNamespace string
+}
+
+// reconciler handles the actual ingress reconciliation logic in response to
+// events.
+type reconciler struct {
+	config Config
+	client client.Client
+}
+
+// Reconcile expects request to refer to a ingress in the operator namespace,
+// and will do all the work to ensure the ingress is in the desired state.
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Info("reconciling", "request", request)
+
+	// Only proceed if we can get the ingress resource.
+	ingress := &configv1.Ingress{}
+	if err := r.client.Get(context.TODO(), request.NamespacedName, ingress); err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("ingress cr not found; reconciliation will be skipped", "request", request)
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("failed to get ingress %q: %v", request, err)
+	}
+
+	// Generate namespaceName mapping
+	mapToSecretName := map[types.NamespacedName]string{}
+	for _, componentRoute := range ingress.Spec.ComponentRoutes {
+		mapToSecretName[types.NamespacedName{Namespace: componentRoute.Namespace, Name: componentRoute.Name}] = componentRoute.ServingCertKeyPairSecret.Name
+	}
+
+	// For each Status.ComponentRoute resource, if a Spec.ComponentRoute with a matching
+	// namespace.name touple exists create the role and roleBinding for each consumer users.
+	// If no matching Spec.ComponentRoute exists, make sure any RBAC that was generated is removed.
+	for _, componentRoute := range ingress.Status.ComponentRoutes {
```

Comment:
you need to go in the other direction.  In order to be able to observe a status and know a thing needs to be removed, you need to find all the roles and rolebindings you've created and see which ones don't belong.  A label seesm appropriate.